From: Shaohua Li <shli@fb.com>
Date: Fri, 23 Jun 2017 19:05:59 +0200
Subject: [PATCH] md: fix a null dereference
References: bsc#1040351
Patch-mainline: submitted linux-raid 2017-06-23

rdev->mddev could be null in start time.

Reported-by: Ming Lei <ming.lei@redhat.com>
Fix: 5a85071c2cbc(md: use a separate bio_set for synchronous IO.)
Cc: NeilBrown <neilb@suse.com>
Signed-off-by: Shaohua Li <shli@fb.com>
Acked-by: Hannes Reinecke <hare@suse.com>
---
 drivers/md/md.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/md/md.c
+++ b/drivers/md/md.c
@@ -203,7 +203,7 @@ struct bio *md_bio_alloc_sync(mddev_t *m
 	struct bio *b;
 	mddev_t **mddevp;
 
-	if (!mddev->sync_set)
+	if (!mddev || !mddev->sync_set)
 		return bio_alloc(GFP_NOIO, 1);
 
 	b = bio_alloc_bioset(GFP_NOIO, 1,
